to_do_app/src/Reducers/todoReducer.js

```diff
@@ -0,0 +1,31 @@
+import { ADD_TODO, TOGGLE_TODO } from "../Actions/actionTypes";
+
+import { addTodo, toggleToDo } from "../Actions/actions";
+//do i need to import the action functions?
+
+//is there where i say what i want the initial state to be? and should it be a todo since this is what we are updating?
+const initialState = {
+  id: 0,
+  text: "",
+  completed: false
+};
+
+const todoReducer = (state = initialState, action) => {
+  //reducers update application state by action types?
+  Object.freeze(state);
+  const newState = { ...state };
+  //do i need to create a newState like this everytime?
+  switch (action.type) {
+    case ADD_TODO:
```

Comment:
In here you would use your `action.payload` which would be a todo object, then you would add this object to your newState, then return the newState where the new todo is part of it.